src/front/wgsl/mod.rs

```diff
@@ -856,13 +855,25 @@ impl Parser {
                     }
                     lexer.expect(Token::Paren(')'))?;
                     if components.is_empty() {
-                        crate::Expression::As {
-                            expr: last_component,
-                            kind: kind.ok_or(Error::BadTypeCast(word))?,
-                            convert: true,
+                        let last_component_inner = ctx.resolve_type(last_component)?;
+                        match (&inner, last_component_inner) {
+                            (crate::TypeInner::Scalar { .. }, crate::TypeInner::Scalar { .. })
+                            | (crate::TypeInner::Matrix { .. }, crate::TypeInner::Matrix { .. })
+                            | (crate::TypeInner::Vector { .. }, crate::TypeInner::Vector { .. }) => {
+                                ctx.types.fetch_or_append(crate::Type { name: None, inner });
```

Comment:
this isn't necessary. If the snapshot is changed, it's totally fine, as long as the produced code is correct. Our CI will validate that it's at least valid.

Reply:
Without `fetch_or_append` there, the msl shader output differs from the snapshots in the tests. The typedef names would change and, in particular, a `typedef int` would be missing in `shadow.msl`, which I don't think is correct?

Example:

```msl
   34    34 │
   35       │-typedef int type9;
         35 │+typedef metal::float3 type9;
   36    36 │
   37       │-typedef metal::float3 type10;
   38       │-
   39       │-typedef bool type11;
         37 │+typedef bool type10;
   40    38 │
```

Reply:
I pushed a new version with adjusted snapshots

Reply:
The CI will yell at you if the typedefs are wrong, so just give it a try